gstd/src/common/handlers.rs

```diff
@@ -25,50 +25,208 @@
 //! debug and non-debug mode, for programs built in `wasm32` architecture.
 //! For `debug` mode it provides more extensive logging.
 
+#[cfg(target_arch = "wasm32")]
 #[cfg(feature = "oom-handler")]
 #[alloc_error_handler]
 pub fn oom(_: core::alloc::Layout) -> ! {
     crate::ext::oom_panic()
 }
 
-#[cfg(feature = "panic-handler")]
-mod panic_handler {
-    use crate::ext;
-    use core::panic::PanicInfo;
+/// We currently support 3 panic handler profiles:
+/// - `panic-handler`: it displays `no info`
+/// - `panic-message`: it displays `'{message}'`
+///   - In nightly Rust, we use `#![feature(panic_info_message)]` and the
+///     [`write!`] macro.
+///   - In stable Rust, we need to modify the default panic handler message
+///     format.
+/// - `panic-location`: it displays `'{message}', {location}`
+///
+/// Default panic handler message format (according to <https://github.com/rust-lang/rust/pull/112849>):
+/// - Rust  <1.73: `panicked at '{message}', {location}`
+/// - Rust >=1.73: `panicked at {location}:\n{message}`
```

Comment:
We can evaluate that in order to compile rust on stable >= 1.73 (add this to lib.rs)

Reply:
It seems to me more convenient to have tests and implementation in one file